### PR TITLE
WinUI guidance update

### PR DESCRIPTION
**Pull request by jamescrosswell**
## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

Integrating user feedback regarding the new WinUI documentation from https://github.com/getsentry/sentry-dotnet/issues/2778#issuecomment-1829559140

### CODE REVIEW

**Comment by bruno-garcia** [comment]
@jamescrosswell there's a conflict

File: src/platforms/dotnet/guides/winui/index.mdx [hunk 1]
```diff
@@ -7,9 +7,9 @@ redirect_from:
 
 Sentry's .NET SDK works with Windows UI Library applications through the [Sentry NuGet package](https://www.nuget.org/packages/Sentry).
 
-## Configuration (for Just-In-Time compilation)
+## Configuration (without trimming)
```
Comment (shanamatthews) [inline src/platforms/dotnet/guides/winui/index.mdx:10]:
```suggestion
## Configuration (Without Trimming)
```

Comment (shanamatthews) [reply src/platforms/dotnet/guides/winui/index.mdx:10]: These headers are actually supposed to be title case. Sorry for missing it last time!

File: src/platforms/dotnet/guides/winui/index.mdx [hunk 2]
```diff
@@ -55,9 +55,9 @@ sealed partial class App : Application
 }
 ```
 
-## Configuration (Ahead-of-Time compilation)
+## Configuration (with trimming)
```
Comment (shanamatthews) [inline src/platforms/dotnet/guides/winui/index.mdx:58]:
```suggestion
## Configuration (With Trimming)
```

**Review by shanamatthews — APPROVED** [review]
👍

**Review by bruno-garcia — APPROVED** [review]

**Pull request merged**